pkg/app/mysql.go

```diff
@@ -0,0 +1,227 @@
+	for i := 0; i < n; i++ {
+		stdout, stderr, err := kube.Exec(mdb.cli, mdb.namespace, podname, containername, insertRecordCMD, nil)
+		format.Log(podname, containername, stdout)
+		format.Log(podname, containername, stderr)
+		if err != nil {
+			log.WithError(err).Print("Error while inserting values into table.")
+			return err
+		}
+	}
+	return nil
+}
+
+func (mdb *MysqlDB) Count(context.Context) (int, error) {
+	podname, containername, err := mdb.GetPodAndContainerName(mdb.namespace)
+	selectRowsCMD := []string{"sh", "-c", "mysql -u root --password=" + mdb.password + " -e 'use testdb; select count(*) from pets; '"}
+	stdout, stderr, err := kube.Exec(mdb.cli, mdb.namespace, podname, containername, selectRowsCMD, nil)
+	format.Log(podname, containername, stdout)
+	format.Log(podname, containername, stderr)
+	if err != nil {
+		log.WithError(err).Print("Error while selecting rows from table.")
+		return 0, err
+	}
+	// get the returned cound and convert it to int, to return
+	rowsReturned, err := strconv.Atoi((strings.Split(stdout, "\n")[1]))
+	return rowsReturned, nil
+}
+
+func (mdb *MysqlDB) Reset(context.Context) error {
+	log.Print("MysqlDB: Resetting the mysql instance.")
+	// delete all the data from the table
+	podname, containername, err := mdb.GetPodAndContainerName(mdb.namespace)
+	deleteFromTableCMD := []string{"sh", "-c", "mysql -u root --password=" + mdb.password + " -e 'DROP DATABASE IF EXISTS testdb'"}
+	stdout, stderr, err := kube.Exec(mdb.cli, mdb.namespace, podname, containername, deleteFromTableCMD, nil)
+	format.Log(podname, containername, stdout)
+	format.Log(podname, containername, stderr)
+	if err != nil {
+		log.WithError(err).Print("Error while dropping table.")
+		return err
+	}
+
+	// create the database and a pets dummy table
+	createTableCMD := []string{"sh", "-c", "mysql -u root --password=" + mdb.password + " -e 'create database testdb; use testdb;  CREATE TABLE pets (name VARCHAR(20), owner VARCHAR(20), species VARCHAR(20), sex CHAR(1), birth DATE, death DATE);'"}
+	stdout, stderr, err = kube.Exec(mdb.cli, mdb.namespace, podname, containername, createTableCMD, nil)
+	format.Log(podname, containername, stdout)
+	format.Log(podname, containername, stderr)
+	if err != nil {
+		log.WithError(err).Print("Error while creatig table, in RESET table.")
+		return err
+	}
+
+	return nil
+}
+
+func (mdb *MysqlDB) ConfigMaps() map[string]crv1alpha1.ObjectReference {
+	return nil
+}
+
+func (mdb *MysqlDB) Secrets() map[string]crv1alpha1.ObjectReference {
+	return map[string]crv1alpha1.ObjectReference{
+		"mysql": crv1alpha1.ObjectReference{
+			Kind:      "Secret",
+			Name:      "mysql-test",
+			Namespace: mdb.namespace,
+		},
+	}
+}
+
+// GetPodAndContianerName takes namespace as input and returns the pod and container that is running in
+// that namespace for deployment that was created through helm
+func (mdb *MysqlDB) GetPodAndContainerName(deploymentNamespace string) (string, string, error) {
```

Comment:
@viveksinghggits We should probably add new helper functions like -
```
StatefulSetContainers(ctx context.Context, kubeCli kubernetes.Interface, namespace string, name string) []string
DeploymentContainers(ctx context.Context, kubeCli kubernetes.Interface, namespace string, name string) []string
```
 in `pkg/kube/workload.go` to return containers from Statefulset or Deployment. Since we will be needing this for other apps also

@tdmanv @SupriyaKasten Any thoughts?

Reply:
Hi @PrasadG193 ,
I have refactored the code to use the new methods from `kube` package.